### ISSUE TITLE
automoduleconfig doesn't pick up configuration in nested structures

### ISSUE THREAD

**Issue by willkg** [willkg/everett#184]
automoduleconfig parses Python files and then pulls configuration out from it. However, I'm pretty sure it only looks at the top level of the AST and doesn't traverse the whole thing.

https://github.com/willkg/everett/blob/c76db94b025e1d24b6868a2f02a6e7d5d2400825/src/everett/sphinxext.py#L654-L666

I was thinking that's fine, but then I was doing a Django update and hit this settings code:

```
CACHES = {
    "default": {
        "BACKEND": "django.core.cache.backends.memcached.PyMemcacheCache",
        "LOCATION": config("CACHE_LOCATION", "127.0.0.1:11211"),
        "TIMEOUT": config("CACHE_TIMEOUT", 500),
        "KEY_PREFIX": config("CACHE_KEY_PREFIX", "socorro"),
    }
}
```

In Socorro, we do this sort of thing with logging and some other things, too. The configuration for `CACHE_LOCATION` won't get picked up with automoduleconfig as it's implemented now. I think this is a valid use case that automoduleconfig should cover.

**Comment by willkg** [comment]
There are a couple of conflated things here.

First is that automoduleconfig needs to walk the AST recursively--not just look at the top level. That's pretty straight-forward to change.

The second is that automoduleconfig needs to support `ast.Assign` nodes (`OPTION = _config(...)`) as well as `ast.Dict` nodes. We'll need to add support for the latter. That's probably doable.

**Comment by willkg** [comment]
Fixed by #193.
